### PR TITLE
fix: env-overrides with tests

### PR DESCRIPTION
**Pull request by adwk67**
# Description

fix: implemented env-var merge and added tests

Tests:
AKS: :green_circle: 
OKD: :green_circle: (except Omid test, but a separate issue has been opened for that)

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
- [x] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] Code contains useful logging statements
- [x] (Integration-)Test cases added
- [x] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: rust/crd/src/lib.rs [hunk 1]
```diff
@@ -655,4 +671,134 @@ impl HbaseCluster {
         tracing::debug!("Merged config: {:?}", conf_rolegroup);
         fragment::validate(conf_rolegroup).context(FragmentValidationFailureSnafu)
     }
+
+    pub fn merged_env(&self, rolegroup_config: Option<&BTreeMap<String, String>>) -> Vec<EnvVar> {
```
Comment (razvan) [inline rust/crd/src/lib.rs:675]: What do you think about renaming it and making it a stand alone function ? For example:

```pub fn make_env (env_map: Option<&BTreeMap<String, String>>) -> Vec<EnvVar>```

**Review by razvan — APPROVED** [review]
lgtm

**Comment by razvan** [comment]
why is the "definition of done" duplicated ? :)

File: rust/crd/src/lib.rs (hunk 1 above)
Comment (adwk67) [reply rust/crd/src/lib.rs]: Good idea! Changed here: 0545c9c

Comment (razvan) [reply rust/crd/src/lib.rs]: I would have also renamed it to `make_env` or something else more general since it knows nothing about merged configuration.

**Review by razvan — APPROVED** [review]

**Pull request merged**